libkbfs/keybase_service_base.go

```diff
@@ -693,7 +723,11 @@ func (k *KeybaseServiceBase) getCurrentSession(
 		ctx, "new session with username %s, uid %s, crypt public key %s, and verifying key %s",
 		s.Name, s.UID, s.CryptPublicKey, s.VerifyingKey)
 
+	k.sessionCacheLock.Lock()
+	defer k.sessionCacheLock.Unlock()
 	k.cachedCurrentSession = s
+	close(k.sessionInProgressCh)
```

Comment:
Actually, can this be nil here?

Reply:
It can't be `nil`, because for `doRPC` above to be `true`, this variable must have been set, and can only be cleared by that same goroutine.  I guess it should be in a `defer` actually, in case there's an error.  Will fix.

Reply:
Why can it only be cleared by the same goroutine? What if getCurrentSession is called by another routine?

Reply:
The other goroutine will be stuck in the `for` loop above until they either see a real session, or until `doRPC` returns `true`.